src/App.tsx

```diff
@@ -7,38 +7,144 @@ import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
+import {
+  createComment,
+  deleteComment,
+  getPostComments,
+  getUserPosts,
+  getUsers,
+} from './api/posts';
+import { User } from './types/User';
+import { Post } from './types/Post';
 import { Loader } from './components/Loader';
+import { Comment, CommentData } from './types/Comment';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [post, setPost] = useState<Post | null>(null);
+  const [comments, setComments] = useState<Comment[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [hasCommentsError, setHasCommentsError] = useState(false);
+  const [commentAdditionError, setCommentAdditionError] = useState(false);
+  const [commentAdditionLoading, setCommentAdditionLoading] = useState(false);
+
+  useEffect(() => {
+    getUsers()
```

Comment:
It's recommended to handle errors when fetching users with try-catch and show appropriate error messages.